pkg/webhook/v1beta1/pod/inject_webhook.go

```diff
@@ -316,3 +324,64 @@ func (s *SidecarInjector) getMetricsCollectorArgs(trial *trialsv1beta1.Trial, me
 
 	return args, nil
 }
+
+// Mutate trial container with shared Suggestions PVC when algorithm settings contains suggestion_trial_dir
+func (s *SidecarInjector) mutateSuggestionVolume(pod *v1.Pod, primaryContainerName string, trial *trialsv1beta1.Trial) error {
+	// Suggestion name == Experiment name
+	// Suggestion namespace == Trial namespace
+	experimentName := trial.ObjectMeta.Labels[consts.LabelExperimentName]
+	experiment := &experimentsv1beta1.Experiment{}
+	if err := s.client.Get(context.TODO(), apitypes.NamespacedName{Name: experimentName, Namespace: trial.Namespace}, experiment); err != nil {
+		return err
+	}
+	suggestion := &suggestionsv1beta1.Suggestion{}
+	if err := s.client.Get(context.TODO(), apitypes.NamespacedName{Name: experimentName, Namespace: trial.Namespace}, suggestion); err != nil {
+		return err
+	}
+
+	// Check if mutation is needed
+	checkpointPath := ""
+	for _, s := range suggestion.Spec.Algorithm.AlgorithmSettings {
+		if s.Name == consts.SuggestionVolumeMountKey && s.Value != "" {
+			checkpointPath = s.Value
+			break
+		}
+	}
+	if checkpointPath == "" {
+		return nil
+	}
+
+	// Generate folder name in format: <ExperimentName>/<TrialName>
+	checkpointFolder := filepath.Join(experimentName, trial.Name)
+
+	// Suggestion volume for the trial to the MetricsCollector
+	suggestionVolume := v1.Volume{
+		Name: consts.ContainerSuggestionVolumeName,
+		VolumeSource: v1.VolumeSource{
+			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+				ClaimName: util.GetSuggestionPersistentVolumeClaimName(suggestion),
+			},
+		},
+	}
+
+	vm := v1.VolumeMount{
+		Name:      suggestionVolume.Name,
+		MountPath: checkpointPath,
+		SubPath:   checkpointFolder,
+	}
+
+	primaryContainerIndex := getPrimaryContainerIndex(pod.Spec.Containers, trial.Spec.PrimaryContainerName)
+	if err := addContainerVolumeMount(&pod.Spec.Containers[primaryContainerIndex], &vm); err != nil {
```

Comment:
I think addContainerVolumeMount() always returns nil. Do we need to check errors?